internal/servers/repository_worker.go

```diff
@@ -292,22 +292,30 @@ func (r *Repository) UpsertWorkerStatus(ctx context.Context, worker *Worker, opt
 				case 1:
 					break
 				default:
-					return errors.New(ctx, errors.MultipleRecords, op, fmt.Sprintf("multiple records found when updating worker with id %q", worker.GetPublicId()))
+					return errors.New(ctx, errors.MultipleRecords, op, fmt.Sprintf("multiple records found when updating worker with id %q", workerClone.GetPublicId()))
 				}
 
-			case worker.GetName() != "":
-				worker.Type = KmsWorkerType.String()
+			case workerClone.GetName() != "":
+				workerClone.Type = KmsWorkerType.String()
 				workerCreateConflict := &db.OnConflict{
 					Target: db.Columns{"public_id"},
-					Action: append(db.SetColumns([]string{"description", "address"}),
+					Action: append(db.SetColumns([]string{"address"}),
 						db.SetColumnValues(map[string]interface{}{"last_status_time": "now()"})...),
 				}
 				var withRowsAffected int64
-				err := w.Create(ctx, worker, db.WithOnConflict(workerCreateConflict), db.WithReturnRowsAffected(&withRowsAffected),
+				err := w.Create(ctx, workerClone, db.WithOnConflict(workerCreateConflict), db.WithReturnRowsAffected(&withRowsAffected),
```

Comment:
Just to check, this will update/override the values in `workerClone` right? Otherwise the check after on Description won't work.

Reply:
Yes, validated with the debugger that it updates workerClone